src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -272,6 +272,13 @@ public static PackageSpec GetPackageSpec(IEnumerable<IMSBuildItem> items)
                     result.RestoreMetadata.RestoreLockProperties = GetRestoreLockProperites(specItem);
                 }
 
+                if (restoreType == ProjectStyle.PackagesConfig)
```

Comment:
Replace these multiple `if` statements with one `switch` statement.

Reply:
If they were if-else if-else blocks, it would be simple and equivalent. However, due to how this method works it would be a big structural change and I'm not confident it would improve readability.

Reply:
See the updated LockFileDependency.cs to see an example of how using switch, rather than if-else, causes duplicate code for bits that are common between different cases. The LockFileDependency.cs example is small and simple, but this MSBuildRestoreUtility.cs code has much more common code that I think would be more difficult using switch, rather than the current if-else.